system/Config/Factories.php

```diff
@@ -424,19 +430,19 @@ public static function reset(?string $component = null)
     {
         if ($component) {
             unset(
-                static::$options[$component],
-                static::$aliases[$component],
-                static::$instances[$component],
-                static::$updated[$component]
+                self::$options[$component],
```

Comment:
If these are changed to `self::`, should the properties be changed to `private`?

Reply:
I think Factories should be `final` - we lean too heavily on it for internals and it isn't available for service location to begin with.

Reply:
Let's merge this with `develop` and make Factories final in `4.5`.

Reply:
I sent PR #8056.